Project03/project03.Rmd

```
@@ -0,0 +1,292 @@
+    return (sequence[N:N+k])
+
+
+
+def fuzzy_diff(x, y, epsilon):
+    """
+    Compares x and y within epsilon difference, and returns true or false
+  
+    Args:
+      x,y (floats):  values to be compared, are they within epislon of each other if so function returns TRUE, otherwise, returns FALSE
+    epsilon (float) - the amount of difference, or tolerance between X & Y that we will allow
+    
+    Returns (boolean): if abs(x-y) os greater than epislon
+    """
+    
+    if (abs(x-y) < epsilon):
+      return(1) # basically the same!
+    else:
+      return(0) # not the same!
+
+
+
+def normalized_scores( poss_scores ):
+  ''' 
+  Converts a list of kmer scores into normalized values, and returns a list of normalzed floats, normalization is 2^(value)/sum(all exp2 values in pos & neg list)
+    
+  Args:
+    poss_scores:  list of floats-contains kmer scores for motifs on DNAi strand
+    
+  Output:
+    returns a list of nornmalized floats (probabilities)
+  '''
+  
+  exp_scores = [math.exp2(score) for score in poss_scores]
+  sum_exp_scores = sum(exp_scores)
+  probabilities = [score/sum_exp_scores for score in exp_scores]
+  
+  return(probabilities)
+
+
+
+def choose_motif(motif_list, weights):
+  '''
+  function choose_motif takes a string of motif values and scores, and selects one in the list randomly, returning the index into the array given.  Scores should come from the calculations on the motif_list, so they need to be of the same length
+  
+  args:
+    motif_list (list of strings): list of motifs 
+    weights (list of floats): list of probabilities of the scores associated with the motifs.
+    
+  Output: string with name of motif that was choosen by our technique
+  '''
+  if (motif_list):
+    if len(motif_list) == len(weights):
+      picked_item = random.choices(motif_list, weights, k=1)[0]  
+      return(picked_item)
+    else:
+      print(f"Error: choose_motif(): {motif_list}, {current_ic} are not the same length")
```

If you're intending to catch an error with this, you might want to actually `raise ValueError` or use try/except rather than just printing -- e.g. you can check at the start of the function with something like `if not motif_list or len(motif_list) != len(weights) ... raise ValueError` (or you can do it separately too)